src/components/container/EhrNewsFeedEventDetail/EhrNewsFeedEventDetail.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState } from 'react';
+import { getNewsFeedPage } from '../../../helpers/news-feed/data';
+import { Action, Card, LoadingIndicator, TextBlock, Title, UnstyledButton } from '../../presentational';
+import StatBlock from '../../presentational/StatBlock';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
+import { format, parseISO } from 'date-fns';
+import { eventTypeDefinitions } from '../../../helpers/news-feed/eventTypeDefinitions';
+import { EhrNewsFeedClaimProcedureModel, EhrNewsFeedClaimServiceModel, EhrNewsFeedEventModel, EhrNewsFeedEventType, EhrNewsFeedType, EhrNewsFeedLabReportModel, EhrNewsFeedProcedureModel } from '../../../helpers/news-feed/types';
+import "./EhrNewsFeedEventDetail.css"
+import language from '../../../helpers/language';
+
+export interface EhrNewsFeedEventDetailProps {
+    feed: EhrNewsFeedType
+    pageId?: string
+    pageDate?: string
+    previewState?: EhrNewsFeedEventType
+    onViewLabObservationTermInfo(labObservationID: string): void
+}
+
+export default function (props: EhrNewsFeedEventDetailProps) {
+    let [loading, setLoading] = useState<boolean>(false);
+    let [event, setEvent] = useState<EhrNewsFeedEventModel | null>(null);
+
+    function load() {
+        if (props.previewState) {
+            let handler = eventTypeDefinitions[props.previewState!];
+            setEvent(handler.getPreviewEvent());
+            return;
+        }
+
+        setLoading(true);
+        getNewsFeedPage(props.feed, props.pageId, props.pageDate).then((result) => {
+            setEvent(result.Events[0]);
+            setLoading(false);
+        });
+    }
+
+    useEffect(() => {
+        load();
+    }, []);
+
+    return <div className="mdhui-ehr-news-feed-event-detail">
+        {loading &&
+            <LoadingIndicator />
+        }
+        {event &&
+            <>
+                <NewsFeedDetailTitle event={event} />
+                {event.Type == "ProcedureGroup" && <ProcedureGroupDetail event={event} />}
+                {event.Type == "ClaimProcedureGroup" && <ClaimProcedureGroupDetail event={event} />}
+                {event.Type == "LabReport" && <LabReportDetail event={event} onViewLabObservationTermInfo={props.onViewLabObservationTermInfo} />}
+                {event.Type == "ClaimServiceGroup" && <ClaimServiceGroupDetail event={event} />}
+            </>
+        }
+    </div>
+}
+
+function NewsFeedDetailTitle(props: { event: EhrNewsFeedEventModel }) {
+    let handler = eventTypeDefinitions[props.event.Type];
+
+    let title = handler.getDetailTitle ? handler.getDetailTitle(props.event) : undefined;
+    if (!title) { return null; }
+
+    let date = parseISO(props.event.Date);
+    let subtitle = `${format(parseISO(props.event.Date), "EEEE, MMMM do, y h:mm a")} • ${props.event.Patient.RecordAuthority}`
```

Comment:
Are we able to localize this for Spanish?

Reply:
Good call, done.  I think we do probably need to do a pass in MyDataHelpsUI to ensure we use locale-specific date formats, like not using MM/dd/yyyy in europe, probably need to rely a bit more on the out of box date formats to achieve that rather than format strings